bundles/org.openhab.core.config.discovery.sddp/src/main/java/org/openhab/core/config/discovery/sddp/SddpDevice.java

```diff
@@ -131,7 +131,7 @@ public SddpDevice(Map<String, String> headers, boolean offline) {
 
         String[] hostParts = host.split("-|_");
         macAddress = hostParts.length <= 1 ? ""
-                : hostParts[hostParts.length - 1].replaceAll("(..)(?!$)", "$1-").toLowerCase();
+                : hostParts[hostParts.length - 1].replaceAll(":", "").replaceAll("(..)(?!$)", "$1-").toLowerCase();
```

Comment:
```suggestion
                : hostParts[hostParts.length - 1].replace(":", "").replaceAll("(..)(?!$)", "$1-").toLowerCase();
```
No regex needed, ```replace``` should be more efficient.